tests/smoketest.sh

```diff
@@ -0,0 +1,145 @@
+    # then, data is trimmed to a determined length to make training
+    # go faster.
+    
+    python3 data_process.py \
+    --data_path="$SAMPLE_DATA_PATH" \
+    --data_output_path="$DATA_DIR" \
+    --max_seq_len=4096 \
+    --model_name_or_path="$MODEL_NAME"
+
+    # trim data so we only keep the first 'n' samples.
+    # should be enough data for training to be meaningful but not enough
+    # that training takes a large amount of time.
+    echo "$(head -"$NUM_SAMPLES_TRAINED_ON" "$COMPUTED_DATA_PATH")" > "$COMPUTED_DATA_PATH"
+
+    echo "TRAINING ON $(wc -l "$COMPUTED_DATA_PATH") SAMPLES"
+}
+
+function _cleanup_saved_checkpoints() {
+    echo "CLEARING CHECKPOINTS: $CHECKPOINTS_DIR"
+    rm -rf "$CHECKPOINTS_DIR"
+    mkdir "$CHECKPOINTS_DIR"
+}
+
+function test_standard_loop () {
+    # Tests most common training parameters.
+    # - FSDP
+    # - Save full state
+    # - Save hf_format
+    # - padding-free
+    # - flash attention
+
+    torchrun \
+    --standalone \
+    --nproc_per_node="$NUM_GPUS" \
+    main_ds.py \
+    --model_name_or_path="$MODEL_NAME" \
+    --data_path="$COMPUTED_DATA_PATH" \
+    --output_dir="$CHECKPOINTS_DIR" \
+    --num_epochs=1 \
+    --effective_batch_size=128 \
+    --save_samples=0 \
+    --checkpoint_at_epoch \
+    --accelerate_full_state_at_epoch \
+    --distributed_training_framework="$DISTRIB_FRAMEWORK" \
+    --max_batch_len="$MAX_BATCH_LEN" \
+    --is_granite \
+}
+
+function test_standard_loop_nongranite () {
+    # Tests most common training parameters without
+    # using Granite
+    # - FSDP
+    # - Save full state
+    # - Save hf_format
+    # - flash attention
+
+    torchrun \
+    --standalone \
+    --nproc_per_node="$NUM_GPUS" \
+    main_ds.py \
+    --model_name_or_path="$MODEL_NAME" \
+    --data_path="$COMPUTED_DATA_PATH" \
+    --output_dir="$CHECKPOINTS_DIR" \
+    --num_epochs=1 \
+    --effective_batch_size=128 \
+    --save_samples=0 \
+    --checkpoint_at_epoch \
+    --accelerate_full_state_at_epoch \
+    --distributed_training_framework="$DISTRIB_FRAMEWORK" \
+    --max_batch_len="$MAX_BATCH_LEN" \
+    # --is_granite \
+}
+
+function test_standard_loop_noflashattention_nogranite () {
+    # Tests most common training parameters without
+    # using Granite or Flash Attention
+    # - FSDP
+    # - Save full state
+    # - Save hf_format
+
+    torchrun \
+    --standalone \
+    --nproc_per_node="$NUM_GPUS" \
+    main_ds.py \
+    --model_name_or_path="$MODEL_NAME" \
+    --data_path="$COMPUTED_DATA_PATH" \
+    --output_dir="$CHECKPOINTS_DIR" \
+    --num_epochs=1 \
+    --effective_batch_size=128 \
+    --save_samples=0 \
+    --checkpoint_at_epoch \
+    --accelerate_full_state_at_epoch \
+    --distributed_training_framework="$DISTRIB_FRAMEWORK" \
+    --max_batch_len="$MAX_BATCH_LEN" \
+    --disable_flash_attn
+    # --is_granite \
+}
+
+# ############### Setup and tests ############### 
+setup_tmpdir
```

Comment:
We should probably move this into a `main` function:

```suggestion
function main() {
    setup_tmpdir
    trap "rm -rf $TMP_DIR" EXIT
    
    #NOTE (jkunstle): script is run as though it's
    # in the same source dir as main_ds and data_process.
    cd "$CORRECT_WORKING_DIR"
    echo "CURRENT WORKING DIRECTORY: $(pwd)"
    
    prepare_data
    test_standard_loop_noflashattention_nogranite
    _cleanup_saved_checkpoints
    test_standard_loop_nongranite
    _cleanup_saved_checkpoints
    test_standard_loop
}

main
```

Reply:
👍